1.10.1 breaks ESLint 2.13.1 and Node < 4

I work with Meteor which is just starting to adopt Node 4 (in Beta). We have many sample apps running Node 0.10.x.

I just downloaded the main Meteor todos app, ran "npm install" and got eslint-plugin-import 1.10.1 installed with ESLint 2.13.1. This produced the following error...

```bash
rhett@dreamshot:~/oob/meteors/todos1.3$ meteor npm run lint

> @ lint /home/rhett/oob/meteors/todos1.3
> eslint .


/home/rhett/oob/meteors/todos1.3/node_modules/eslint-plugin-import/lib/index.js:6
const rules = exports.rules = {
^^^^^

Oops! Something went wrong! :(

ESLint couldn't find the plugin "eslint-plugin-import". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-import is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-import@latest --save-dev

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.


npm ERR! Linux 4.4.0-28-generic
npm ERR! argv "node" "/home/rhett/.meteor/packages/meteor-tool/.1.3.0_3.139i3kk++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/npm" "run" "lint"
npm ERR! node v0.10.43
npm ERR! npm  v2.14.22
npm ERR! code ELIFECYCLE
npm ERR! @ lint: `eslint .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ lint script 'eslint .'.
npm ERR! This is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs 
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls 
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/rhett/oob/meteors/todos1.3/npm-debug.log
```

I'm no NPM expert, but do you perhaps need a node 4 requirement in your package.json now?
```json
"engines": {
    "node": ">=4"
  }
```

Reverting to 1.10.0 fixed my problem. Also, running with Node 4 fixed it, but that isn't something our user base at large is ready for yet.

Thank you! That will stop problems with those of our users who have not yet moved up to our latest version. I verified that the Meteor 1.3 products that use Node v0.10.43 and npm  v2.14.22 are happy with eslint 2.13.1 and eslint-plugin-import 1.10.2.

The narrower issue in my second comment does, however, persist. This seems to be a factor of installing with NPM v3.9.6 and running with NPM v3.9.6 and Node v0.10.43. If I run with Node v4.4.7 instead, even without reinstalling, the problem goes away. This effects anyone using the latest released version of Meteor, v1.3.4.1.

Looking back, I see that I probably should have given you the output, not the npm-debug.log. Here is the output running with eslint-plugin-import 1.10.2 that indicates an issue at eslint-plugin-import/lib/rules/export.js:93:149

```
rhett@dreamshot:~/oob/meteors/todos1.3.4.1$ ../mymeteor npm run lint
Meteor bundle is correct for 1.3.4.1: packages/meteor-tool/1.3.4_1/mt-os.linux.x86_64/meteor

> @ lint /home/rhett/oob/meteors/todos1.3.4.1
> eslint .

Property 'Symbol(Symbol.iterator)_9.nz0k7st2kbe53ik9' of object [object Map] is not a function
TypeError: Property 'Symbol(Symbol.iterator)_9.nz0k7st2kbe53ik9' of object [object Map] is not a function
    at EventEmitter.ProgramExit (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint-plugin-import/lib/rules/export.js:93:149)
    at EventEmitter.emit (events.js:117:20)
    at NodeEventGenerator.leaveNode (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint/lib/util/node-event-generator.js:49:22)
    at CodePathAnalyzer.leaveNode (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:627:23)
    at CommentEventGenerator.leaveNode (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint/lib/util/comment-event-generator.js:110:23)
    at Controller.traverser.traverse.leave (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint/lib/eslint.js:908:36)
    at Controller.__execute (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/estraverse/estraverse.js:491:28)
    at Controller.Traverser.controller.traverse (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint/lib/util/traverser.js:36:33)
    at EventEmitter.module.exports.api.verify (/home/rhett/oob/meteors/todos1.3.4.1/node_modules/eslint/lib/eslint.js:902:23)

npm ERR! Linux 4.4.0-28-generic
npm ERR! argv "node" "/home/rhett/.meteor/packages/meteor-tool/.1.3.4_1.tskw1g++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/npm" "run" "lint"
npm ERR! node v0.10.45
npm ERR! npm  v3.9.6
npm ERR! code ELIFECYCLE
npm ERR! @ lint: `eslint .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ lint script 'eslint .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs 
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls 
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/rhett/oob/meteors/todos1.3.4.1/npm-debug.log
```

Just to be sure... I just duplicated this completely outside of the meteor environment with a very stripped down package.json and basically empty .js file as follows:

```
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ ls -l
total 8
-rw-rw-r-- 1 rhett rhett 558 Jul  5 14:01 package.json
-rw-rw-r-- 1 rhett rhett  14 Jul  5 13:56 sample.js
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ cat package.json 
{
  "scripts": {
    "lint": "eslint ."
  },
  "dependencies": {
  },
  "devDependencies": {
    "eslint": "^2.13.1",
    "eslint-config-airbnb": "^9.0.1",
    "eslint-import-resolver-meteor": "^0.2.4",
    "eslint-plugin-import": ">1.10.1",
    "eslint-plugin-jsx-a11y": "^1.5.3",
    "eslint-plugin-meteor": "^3.6.0",
    "eslint-plugin-react": "^5.2.2"
  },
  "eslintConfig": {
    "plugins": [
      "meteor"
    ],
    "extends": [
      "airbnb",
      "plugin:meteor/recommended"
    ],
    "settings": {
      "import/resolver": "meteor"
    }
  }
}
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ cat sample.js 
// empty file
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ npm version
{ npm: '3.9.6',
  ares: '1.9.0-DEV',
  http_parser: '1.2',
  modules: '11',
  node: '0.10.45',
  openssl: '1.0.1t',
  uv: '0.10.36',
  v8: '3.14.5.9',
  zlib: '1.2.8' }
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ npm install
/home/rhett/oob/meteors/debug-eslint-plugin-import
├─┬ eslint@2.13.1 
│ ├─┬ chalk@1.1.3 
│ │ ├── ansi-styles@2.2.1 
│ │ ├── escape-string-regexp@1.0.5 
│ │ ├── has-ansi@2.0.0 
│ │ ├── strip-ansi@3.0.1 
│ │ └── supports-color@2.0.0 
│ ├─┬ concat-stream@1.5.1 
│ │ ├── inherits@2.0.1 
│ │ ├─┬ readable-stream@2.0.6 
│ │ │ ├── core-util-is@1.0.2 
│ │ │ ├── process-nextick-args@1.0.7 
│ │ │ ├── string_decoder@0.10.31 
│ │ │ └── util-deprecate@1.0.2 
│ │ └── typedarray@0.0.6 
│ ├─┬ debug@2.2.0 
│ │ └── ms@0.7.1 
│ ├─┬ doctrine@1.2.2 
│ │ ├── esutils@1.1.6 
│ │ └── isarray@1.0.0 
│ ├─┬ es6-map@0.1.4 
│ │ ├── d@0.1.1 
│ │ ├── es5-ext@0.10.12 
│ │ ├── es6-iterator@2.0.0 
│ │ └── event-emitter@0.3.4 
│ ├─┬ escope@3.6.0 
│ │ ├── es6-weak-map@2.0.1 
│ │ └─┬ esrecurse@4.1.0 
│ │   └── estraverse@4.1.1 
│ ├─┬ espree@3.1.6 
│ │ ├── acorn@3.2.0 
│ │ └── acorn-jsx@3.0.1 
│ ├── estraverse@4.2.0 
│ ├── esutils@2.0.2 
│ ├─┬ file-entry-cache@1.2.4 
│ │ └─┬ flat-cache@1.0.10 
│ │   ├─┬ del@2.2.1 
│ │   │ ├─┬ globby@5.0.0 
│ │   │ │ ├─┬ array-union@1.0.2 
│ │   │ │ │ └── array-uniq@1.0.3 
│ │   │ │ └── arrify@1.0.1 
│ │   │ ├── is-path-cwd@1.0.0 
│ │   │ ├─┬ is-path-in-cwd@1.0.0 
│ │   │ │ └── is-path-inside@1.0.0 
│ │   │ ├── pify@2.3.0 
│ │   │ └── rimraf@2.5.3 
│ │   ├── graceful-fs@4.1.4 
│ │   ├── read-json-sync@1.1.1 
│ │   └── write@0.2.1 
│ ├─┬ glob@7.0.5 
│ │ ├── fs.realpath@1.0.0 
│ │ ├─┬ inflight@1.0.5 
│ │ │ └── wrappy@1.0.2 
│ │ ├─┬ minimatch@3.0.2 
│ │ │ └─┬ brace-expansion@1.1.5 
│ │ │   ├── balanced-match@0.4.1 
│ │ │   └── concat-map@0.0.1 
│ │ └── once@1.3.3 
│ ├── globals@9.9.0 
│ ├── ignore@3.1.3 
│ ├── imurmurhash@0.1.4 
│ ├─┬ inquirer@0.12.0 
│ │ ├── ansi-escapes@1.4.0 
│ │ ├── ansi-regex@2.0.0 
│ │ ├─┬ cli-cursor@1.0.2 
│ │ │ └─┬ restore-cursor@1.0.1 
│ │ │   ├── exit-hook@1.1.1 
│ │ │   └── onetime@1.1.0 
│ │ ├── cli-width@2.1.0 
│ │ ├── figures@1.7.0 
│ │ ├─┬ readline2@1.0.1 
│ │ │ ├─┬ code-point-at@1.0.0 
│ │ │ │ └── number-is-nan@1.0.0 
│ │ │ ├── is-fullwidth-code-point@1.0.0 
│ │ │ └── mute-stream@0.0.5 
│ │ ├── run-async@0.1.0 
│ │ ├── rx-lite@3.1.2 
│ │ ├── string-width@1.0.1 
│ │ └── through@2.3.8 
│ ├─┬ is-my-json-valid@2.13.1 
│ │ ├── generate-function@2.0.0 
│ │ ├─┬ generate-object-property@1.2.0 
│ │ │ └── is-property@1.0.2 
│ │ ├── jsonpointer@2.0.0 
│ │ └── xtend@4.0.1 
│ ├─┬ is-resolvable@1.0.0 
│ │ └── tryit@1.0.2 
│ ├─┬ js-yaml@3.6.1 
│ │ ├─┬ argparse@1.0.7 
│ │ │ └── sprintf-js@1.0.3 
│ │ └── esprima@2.7.2 
│ ├─┬ json-stable-stringify@1.0.1 
│ │ └── jsonify@0.0.0 
│ ├─┬ levn@0.3.0 
│ │ ├── prelude-ls@1.1.2 
│ │ └── type-check@0.3.2 
│ ├── lodash@4.13.1 
│ ├─┬ mkdirp@0.5.1 
│ │ └── minimist@0.0.8 
│ ├─┬ optionator@0.8.1 
│ │ ├── deep-is@0.1.3 
│ │ ├── fast-levenshtein@1.1.3 
│ │ └── wordwrap@1.0.0 
│ ├── path-is-absolute@1.0.0 
│ ├── path-is-inside@1.0.1 
│ ├── pluralize@1.2.1 
│ ├── progress@1.1.8 
│ ├─┬ require-uncached@1.0.2 
│ │ ├─┬ caller-path@0.1.0 
│ │ │ └── callsites@0.2.0 
│ │ └── resolve-from@1.0.1 
│ ├── shelljs@0.6.0 
│ ├── strip-json-comments@1.0.4 
│ ├─┬ table@3.7.8 
│ │ ├── bluebird@3.4.1 
│ │ ├── slice-ansi@0.0.4 
│ │ ├── tv4@1.2.7 
│ │ └── xregexp@3.1.1 
│ ├── text-table@0.2.0 
│ └─┬ user-home@2.0.0 
│   └── os-homedir@1.0.1 
├─┬ eslint-config-airbnb@9.0.1 
│ └── eslint-config-airbnb-base@3.0.1 
├─┬ eslint-import-resolver-meteor@0.2.4 
│ ├── object-assign@4.1.0 
│ └── resolve@1.1.7 
├─┬ eslint-plugin-import@1.10.2 
│ ├── builtin-modules@1.1.1 
│ ├── contains-path@0.1.0 
│ ├── es6-set@0.1.4 
│ ├── es6-symbol@3.1.0 
│ ├── eslint-import-resolver-node@0.2.1 
│ ├─┬ lodash.cond@4.4.0 
│ │ ├─┬ lodash._baseiteratee@4.7.0 
│ │ │ └── lodash._stringtopath@4.8.0 
│ │ └── lodash.rest@4.0.3 
│ ├─┬ lodash.endswith@4.1.0 
│ │ ├── lodash._basetostring@4.12.0 
│ │ └── lodash.tostring@4.1.3 
│ ├─┬ lodash.find@4.4.0 
│ │ ├── lodash._baseeach@4.1.3 
│ │ ├── lodash._basefind@3.0.0 
│ │ └── lodash._basefindindex@3.6.0 
│ ├── lodash.findindex@4.4.0 
│ ├─┬ pkg-dir@1.0.0 
│ │ └─┬ find-up@1.1.2 
│ │   ├── path-exists@2.1.0 
│ │   └─┬ pinkie-promise@2.0.1 
│ │     └── pinkie@2.0.4 
│ └── pkg-up@1.0.0 
├─┬ eslint-plugin-jsx-a11y@1.5.3 
│ ├── damerau-levenshtein@1.0.0 
│ └── jsx-ast-utils@1.2.1 
├─┬ eslint-plugin-meteor@3.6.0 
│ ├─┬ babel-polyfill@6.9.0 
│ │ ├── babel-regenerator-runtime@6.5.0 
│ │ ├─┬ babel-runtime@6.9.2 
│ │ │ └── regenerator-runtime@0.9.5 
│ │ └── core-js@2.4.0 
│ ├─┬ babel-register@6.8.0 
│ │ ├─┬ babel-core@6.10.4 
│ │ │ ├─┬ babel-code-frame@6.11.0 
│ │ │ │ └── js-tokens@2.0.0 
│ │ │ ├─┬ babel-generator@6.11.0 
│ │ │ │ ├── babel-runtime@6.9.2 
│ │ │ │ └─┬ detect-indent@3.0.1 
│ │ │ │   ├── get-stdin@4.0.1 
│ │ │ │   ├── minimist@1.2.0 
│ │ │ │   └─┬ repeating@1.1.3 
│ │ │ │     └── is-finite@1.0.1 
│ │ │ ├── babel-helpers@6.8.0 
│ │ │ ├── babel-messages@6.8.0 
│ │ │ ├── babel-register@6.9.0 
│ │ │ ├── babel-runtime@6.9.2 
│ │ │ ├─┬ babel-template@6.9.0 
│ │ │ │ └── babel-runtime@6.9.2 
│ │ │ ├─┬ babel-traverse@6.10.4 
│ │ │ │ ├── babel-runtime@6.9.2 
│ │ │ │ └── globals@8.18.0 
│ │ │ ├─┬ babel-types@6.11.1 
│ │ │ │ ├── babel-runtime@6.9.2 
│ │ │ │ └── to-fast-properties@1.0.2 
│ │ │ ├── babylon@6.8.3 
│ │ │ ├── convert-source-map@1.2.0 
│ │ │ ├── json5@0.4.0 
│ │ │ ├── path-exists@1.0.0 
│ │ │ ├── private@0.1.6 
│ │ │ ├── shebang-regex@1.0.0 
│ │ │ ├── slash@1.0.0 
│ │ │ └── source-map@0.5.6 
│ │ ├─┬ home-or-tmp@1.0.0 
│ │ │ ├── os-tmpdir@1.0.1 
│ │ │ └── user-home@1.1.1 
│ │ ├── lodash@3.10.1 
│ │ ├── path-exists@1.0.0 
│ │ └─┬ source-map-support@0.2.10 
│ │   └─┬ source-map@0.1.32 
│ │     └── amdefine@1.0.0 
│ ├── babel-runtime@6.6.1 
│ ├─┬ invariant@2.2.1 
│ │ └─┬ loose-envify@1.2.0 
│ │   └── js-tokens@1.0.3 
│ ├─┬ lodash.memoize@4.1.0 
│ │ └── lodash._root@3.0.1 
│ └── path-exists@3.0.0 
└── eslint-plugin-react@5.2.2 

npm WARN debug-eslint-plugin-import No description
npm WARN debug-eslint-plugin-import No repository field.
npm WARN debug-eslint-plugin-import No license field.
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ npm run lint

> @ lint /home/rhett/oob/meteors/debug-eslint-plugin-import
> eslint .

Property 'Symbol(Symbol.iterator)_9.2jbx8n5fx2do0f6r' of object [object Map] is not a function
TypeError: Property 'Symbol(Symbol.iterator)_9.2jbx8n5fx2do0f6r' of object [object Map] is not a function
    at EventEmitter.ProgramExit (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint-plugin-import/lib/rules/export.js:93:149)
    at EventEmitter.emit (events.js:117:20)
    at NodeEventGenerator.leaveNode (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint/lib/util/node-event-generator.js:49:22)
    at CodePathAnalyzer.leaveNode (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:627:23)
    at CommentEventGenerator.leaveNode (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint/lib/util/comment-event-generator.js:110:23)
    at Controller.traverser.traverse.leave (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint/lib/eslint.js:908:36)
    at Controller.__execute (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/estraverse/estraverse.js:491:28)
    at Controller.Traverser.controller.traverse (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint/lib/util/traverser.js:36:33)
    at EventEmitter.module.exports.api.verify (/home/rhett/oob/meteors/debug-eslint-plugin-import/node_modules/eslint/lib/eslint.js:902:23)

npm ERR! Linux 4.4.0-28-generic
npm ERR! argv "node" "/home/rhett/.nvm/v0.10.45/bin/npm" "run" "lint"
npm ERR! node v0.10.45
npm ERR! npm  v3.9.6
npm ERR! code ELIFECYCLE
npm ERR! @ lint: `eslint .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ lint script 'eslint .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs 
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls 
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/rhett/oob/meteors/debug-eslint-plugin-import/npm-debug.log
```

> Based on the logs you're using npm v2 so you have two copies of the plugin.

Oddly enough. It works with npm 2, not npm 3. Also, if I npm install with 2 and run with 3, it will work. Only npm install with 3 and run with 3 breaks. Look closer  at the most recent log comments above...
```
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ ls -l
total 8
-rw-rw-r-- 1 rhett rhett 558 Jul  5 14:01 package.json
-rw-rw-r-- 1 rhett rhett  14 Jul  5 13:56 sample.js
```

Note no node-modules installed,,, then

...

rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ npm version
{ npm: '3.9.6',
  ares: '1.9.0-DEV',
  http_parser: '1.2',
  modules: '11',
  node: '0.10.45',
  openssl: '1.0.1t',
  uv: '0.10.36',
  v8: '3.14.5.9',
  zlib: '1.2.8' }
rhett@dreamshot:~/oob/meteors/debug-eslint-plugin-import$ npm install

...

npm ERR! Linux 4.4.0-28-generic
npm ERR! argv "node" "/home/rhett/.nvm/v0.10.45/bin/npm" "run" "lint"
npm ERR! node v0.10.45
npm ERR! npm  v3.9.6
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,7 +5,8 @@ This change log adheres to standards from [Keep a CHANGELOG](http://keepachangel
 
 ## [Unreleased]
 ### Fixed
-- pinned `es6-symbol` dependency to `^3.1.0` instead of `*` 😳 ([#415])
+- removing `Symbol` dependencies (i.e. `for-of` loops) due to Node 0.10 polyfill
+  issue (see [#415]). Should not make any discernible semantic difference.
 
 ## [1.10.2] - 2016-07-04
 ### Fixed

--- a/package.json
+++ b/package.json
@@ -70,7 +70,6 @@
     "doctrine": "1.2.x",
     "es6-map": "^0.1.3",
     "es6-set": "^0.1.4",
-    "es6-symbol": "^3.1.0",
     "eslint-import-resolver-node": "^0.2.0",
     "lodash.cond": "^4.3.0",
     "lodash.endswith": "^4.0.1",

--- a/src/core/getExports.js
+++ b/src/core/getExports.js
@@ -236,18 +236,19 @@ export default class ExportMap {
     if (this.reexports.has(name)) return true
 
     // default exports must be explicitly re-exported (#328)
+    let foundInnerMapName = false
     if (name !== 'default') {
-      for (let dep of this.dependencies.values()) {
-        let innerMap = dep()
+      this.dependencies.forEach((dep) => {
+        if (!foundInnerMapName) {
+          let innerMap = dep()
 
-        // todo: report as unresolved?
-        if (!innerMap) continue
-
-        if (innerMap.has(name)) return true
-      }
+          // todo: report as unresolved?
+          if (innerMap && innerMap.has(name)) foundInnerMapName = true
+        }
+      })
     }
 
-    return false
+    return foundInnerMapName
   }
 
   /**
@@ -276,24 +277,29 @@ export default class ExportMap {
 
 
     // default exports must be explicitly re-exported (#328)
+    let returnValue = { found: false, path: [this] }
     if (name !== 'default') {
-      for (let dep of this.dependencies.values()) {
-        let innerMap = dep()
-        // todo: report as unresolved?
-        if (!innerMap) continue
+      this.dependencies.forEach((dep) => {
+        if (!returnValue.found) {
+          let innerMap = dep()
+          // todo: report as unresolved?
+          if (innerMap) {
 
-        // safeguard against cycles
-        if (innerMap.path === this.path) continue
+            // safeguard against cycles
+            if (innerMap.path !== this.path) {
 
-        let innerValue = innerMap.hasDeep(name)
-        if (innerValue.found) {
-          innerValue.path.unshift(this)
-          return innerValue
+              let innerValue = innerMap.hasDeep(name)
+              if (innerValue.found) {
+                innerValue.path.unshift(this)
+                returnValue = innerValue
+              }
+            }
+          }
         }
-      }
+      })
     }
 
-    return { found: false, path: [this] }
+    return returnValue
   }
 
   get(name) {
@@ -313,21 +319,26 @@ export default class ExportMap {
     }
 
     // default exports must be explicitly re-exported (#328)
+    let returnValue = undefined
     if (name !== 'default') {
-      for (let dep of this.dependencies.values()) {
-        let innerMap = dep()
-        // todo: report as unresolved?
-        if (!innerMap) continue
+      this.dependencies.foreach((dep) => {
+        if (returnValue === undefined) {
+          let innerMap = dep()
+          // todo: report as unresolved?
+          if (innerMap) {
 
-        // safeguard against cycles
-        if (innerMap.path === this.path) continue
+            // safeguard against cycles
+            if (innerMap.path !== this.path) {
 
-        let innerValue = innerMap.get(name)
-        if (innerValue !== undefined) return innerValue
-      }
+              let innerValue = innerMap.get(name)
+              if (innerValue !== undefined) returnValue = innerValue
+            }
+          }
+        }
+      })
     }
 
-    return undefined
+    return returnValue
   }
 
   forEach(callback, thisArg) {

--- a/src/core/getExports.js
+++ b/src/core/getExports.js
@@ -1,4 +1,3 @@
-import 'es6-symbol/implement'
 import Map from 'es6-map'
 
 import * as fs from 'fs'

--- a/src/core/resolve.js
+++ b/src/core/resolve.js
@@ -112,23 +112,24 @@ function fullResolve(modulePath, sourceFile, settings) {
 
   const resolvers = resolverReducer(configResolvers, new Map())
 
-  for (let [name, config] of resolvers) {
-    const resolver = requireResolver(name, sourceFile)
-        , resolved = withResolver(resolver, config)
+  let resolved = { found: false }
+  resolvers.forEach(function (config, name)  {
+    if (!resolved.found) {
+      const resolver = requireResolver(name, sourceFile)
+      resolved = withResolver(resolver, config)
+      if (resolved.found) {
+        // resolvers imply file existence, this double-check just ensures the case matches
+        if (fileExistsWithCaseSync(resolved.path, cacheSettings)) {
+          // else, counts
+          cache(resolved.path)
+        } else {
+          resolved = { found: false }
+        }
+      }
+    }
+  })
 
-    if (!resolved.found) continue
-
-    // resolvers imply file existence, this double-check just ensures the case matches
-    if (!fileExistsWithCaseSync(resolved.path, cacheSettings)) continue
-
-    // else, counts
-    cache(resolved.path)
-    return resolved
-  }
-
-  // failed
-  // cache(undefined)
-  return { found: false }
+  return resolved
 }
 
 function resolverReducer(resolvers, map) {

--- a/src/core/resolve.js
+++ b/src/core/resolve.js
@@ -1,4 +1,3 @@
-import 'es6-symbol/implement'
 import Map from 'es6-map'
 import Set from 'es6-set'
 import assign from 'object-assign'

--- a/src/rules/export.js
+++ b/src/rules/export.js
@@ -1,4 +1,3 @@
-import 'es6-symbol/implement'
 import Map from 'es6-map'
 import Set from 'es6-set'
 
@@ -32,11 +31,11 @@ module.exports = function (context) {
         addNamed(node.declaration.id.name, node.declaration.id)
       }
 
-      if (node.declaration.declarations != null) {
-        for (let declaration of node.declaration.declarations) {
-          recursivePatternCapture(declaration.id, v => addNamed(v.name, v))
-        }
-      }
+      if (node.declaration.declarations == null) return
+
+      node.declaration.declarations.forEach(declaration => {
+        recursivePatternCapture(declaration.id, v => addNamed(v.name, v))
+      })
     },
 
     'ExportAllDeclaration': function (node) {
@@ -62,15 +61,15 @@ module.exports = function (context) {
     },
 
     'Program:exit': function () {
-      for (let [name, nodes] of named) {
-        if (nodes.size <= 1) continue
+      named.forEach((nodes, name) => {
+        if (nodes.size <= 1) return
 
-        for (let node of nodes) {
+        nodes.forEach(node => {
           if (name === 'default') {
             context.report(node, 'Multiple default exports.')
           } else context.report(node, `Multiple exports of name '${name}'.`)
-        }
-      }
+        })
+      })
     },
   }
 }

--- a/src/rules/namespace.js
+++ b/src/rules/namespace.js
@@ -32,7 +32,7 @@ module.exports = function (context) {
           return
         }
 
-        for (let specifier of declaration.specifiers) {
+        declaration.specifiers.forEach((specifier) => {
           switch (specifier.type) {
             case 'ImportNamespaceSpecifier':
               if (!imports.size) {
@@ -51,7 +51,7 @@ module.exports = function (context) {
               break
             }
           }
-        }
+        })
       }
       body.forEach(processBodyStatement)
     },
@@ -129,28 +129,23 @@ module.exports = function (context) {
 
         if (pattern.type !== 'ObjectPattern') return
 
-        for (let property of pattern.properties) {
-
+        pattern.properties.forEach((property) => {
           if (property.key.type !== 'Identifier') {
             context.report({
               node: property,
               message: 'Only destructure top-level names.',
             })
-            continue
-          }
-
-          if (!namespace.has(property.key.name)) {
+          } else if (!namespace.has(property.key.name)) {
             context.report({
               node: property,
               message: makeMessage(property.key, path),
             })
-            continue
+          } else {
+            path.push(property.key.name)
+            testKey(property.value, namespace.get(property.key.name).namespace, path)
+            path.pop()
           }
-
-          path.push(property.key.name)
-          testKey(property.value, namespace.get(property.key.name).namespace, path)
-          path.pop()
-        }
+        })
       }
 
       testKey(id, namespaces.get(init.name))

--- a/src/rules/namespace.js
+++ b/src/rules/namespace.js
@@ -1,4 +1,3 @@
-import 'es6-symbol/implement'
 import Map from 'es6-map'
 
 import Exports from '../core/getExports'

--- a/src/rules/no-duplicates.js
+++ b/src/rules/no-duplicates.js
@@ -1,4 +1,3 @@
-import 'es6-symbol/implement'
 import Map from 'es6-map'
 import Set from 'es6-set'
 

--- a/src/rules/no-duplicates.js
+++ b/src/rules/no-duplicates.js
@@ -5,13 +5,13 @@ import Set from 'es6-set'
 import resolve from '../core/resolve'
 
 function checkImports(imported, context) {
-  for (let [module, nodes] of imported.entries()) {
+  imported.forEach((nodes, module) => {
     if (nodes.size > 1) {
-      for (let node of nodes) {
+      nodes.forEach((node) => {
         context.report(node, `'${module}' imported multiple times.`)
-      }
+      })
     }
-  }
+  })
 }
 
 module.exports = function (context) {

--- a/src/rules/no-mutable-exports.js
+++ b/src/rules/no-mutable-exports.js
@@ -7,15 +7,15 @@ module.exports = function (context) {
   }
 
   function checkDeclarationsInScope({variables}, name) {
-    for (let variable of variables) {
+    variables.forEach((variable) => {
       if (variable.name === name) {
-        for (let def of variable.defs) {
+        variable.defs.forEach((def) => {
           if (def.type === 'Variable') {
             checkDeclaration(def.parent)
           }
-        }
+        })
       }
-    }
+    })
   }
 
   function handleExportDefault(node) {
@@ -32,9 +32,9 @@ module.exports = function (context) {
     if (node.declaration)  {
       checkDeclaration(node.declaration)
     } else if (!node.source) {
-      for (let specifier of node.specifiers) {
+      node.specifiers.forEach((specifier) => {
         checkDeclarationsInScope(scope, specifier.local.name)
-      }
+      })
     }
   }
 

--- a/src/rules/no-named-as-default-member.js
+++ b/src/rules/no-named-as-default-member.js
@@ -57,10 +57,11 @@ module.exports = function(context) {
     if (!isDestructure) return
 
     const objectName = node.init.name
-    for (const { key } of node.id.properties) {
-      if (key == null) continue  // true for rest properties
-      storePropertyLookup(objectName, key.name, key)
-    }
+    node.id.properties.forEach(({key}) => {
+      if (key != null) { // rest properties are null
+        storePropertyLookup(objectName, key.name, key)
+      }
+    })
   }
 
   function handleProgramExit() {
@@ -68,19 +69,19 @@ module.exports = function(context) {
       const fileImport = fileImports.get(objectName)
       if (fileImport == null) return
 
-      for (const {propName, node} of lookups) {
-        if (!fileImport.exportMap.namespace.has(propName)) continue
-
-        context.report({
-          node,
-          message: (
-            `Caution: \`${objectName}\` also has a named export ` +
-            `\`${propName}\`. Check if you meant to write ` +
-            `\`import {${propName}} from '${fileImport.sourcePath}'\` ` +
-            'instead.'
-          ),
-        })
-      }
+      lookups.forEach(({propName, node}) => {
+        if (fileImport.exportMap.namespace.has(propName)) {
+          context.report({
+            node,
+            message: (
+              `Caution: \`${objectName}\` also has a named export ` +
+              `\`${propName}\`. Check if you meant to write ` +
+              `\`import {${propName}} from '${fileImport.sourcePath}'\` ` +
+              'instead.'
+            ),
+          })
+        }
+      })
     })
   }
 

--- a/src/rules/no-named-as-default-member.js
+++ b/src/rules/no-named-as-default-member.js
@@ -5,7 +5,6 @@
  * See LICENSE in root directory for full license.
  */
 
-import 'es6-symbol/implement'
 import Map from 'es6-map'
 
 import Exports from '../core/getExports'

--- a/src/rules/no-unresolved.js
+++ b/src/rules/no-unresolved.js
@@ -3,8 +3,6 @@
  * @author Ben Mosher
  */
 
-import 'es6-symbol/implement'
-
 import resolve from '../core/resolve'
 
 module.exports = function (context) {

--- a/src/rules/no-unresolved.js
+++ b/src/rules/no-unresolved.js
@@ -53,15 +53,17 @@ module.exports = function (context) {
     const modules = call.arguments[0]
     if (modules.type !== 'ArrayExpression') return
 
-    for (let element of modules.elements) {
-      if (element.type !== 'Literal') continue
-      if (typeof element.value !== 'string') continue
+    modules.elements.forEach((element) => {
+      if (element.type === 'Literal' &&
+          typeof element.value === 'string') {
 
-      if (element.value === 'require' ||
-          element.value === 'exports') continue // magic modules: http://git.io/vByan
-
-      checkSourceValue(element)
-    }
+        // magic modules: http://git.io/vByan
+        if (element.value !== 'require' &&
+            element.value !== 'exports') {
+          checkSourceValue(element)
+        }
+      }
+    })
   }
 
   const visitors = {

--- a/tests/src/package.js
+++ b/tests/src/package.js
@@ -1,5 +1,3 @@
-import 'es6-symbol/implement'
-
 var expect = require('chai').expect
 
 var path = require('path')
@@ -36,10 +34,10 @@ describe('package', function () {
   it('exports all configs', function (done) {
     fs.readdir(path.join(process.cwd(), 'config'), function (err, files) {
       if (err) { done(err); return }
-      for (let file of files) {
-        if (file[0] === '.') continue
+      files.forEach(file => {
+        if (file[0] === '.') return
         expect(module.configs).to.have.property(file.slice(0, -3)) // drop '.js'
-      }
+      })
       done()
     })
   })

